src/Util/Writer/DocWriter.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Util\Writer;
+
+use Friendica\Core\Renderer;
+use Friendica\Database\Definition\DbaDefinition;
+use Friendica\Network\HTTPException\ServiceUnavailableException;
+
+/**
+ * Utility class to write content into the '/doc' directory
+ */
+class DocWriter
+{
+	/**
+	 * Creates all database definitions as Markdown fields and create the mkdoc config file.
+	 *
+	 * @param DbaDefinition $definition The Database definition class
+	 * @param string        $basePath   The basepath of Friendica
+	 *
```

Comment:
You seem to add extra empty line here, I mostly removed them. So how do we want to do this here to avoid contradicting edits?

Reply:
That's the way I'm working as well. Since each block is different, we should keep them separated.

Reply:
Okay, I will watch out for this in the future.